File: Solutions/1700-1799/1701. Average Waiting Time/Solution.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        time, wait = 0, 0
        res = wait
        for arrival, cook in customers:
            if time < arrival:
                time = arrival
            time += cook
            wait += time - arrival

        return wait/len(customers)

File: Solutions/1700-1799/1701. Average Waiting Time/Solution.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        int k=0;
        double time=0;
        for(int i=0;i<customers.length;i++)
        {
            int startTime=customers[i][0];
            int serviceTime=customers[i][1];
            
            int serviceStart=Math.max(startTime,k);
            int serviceEnd=serviceStart+serviceTime;
                
            time+=(serviceEnd-startTime);
            k=serviceEnd;
        }
        return (time/customers.length);
    }
}
